design_docs/variable_mapping_reorg.md

```diff
@@ -0,0 +1,184 @@
+results to those from the develop branch for typical test cases on LANL IC
+and Edison.
+
+<h1> Requirements </h1>
+
+<h2> Requirement: mpas_xarray does not include MPAS-Analysis specific
+functionality <br>
+Date last modified: 2017/02/10 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+MPAS-Analysis specific functionality such as variable mapping should be 
+removed from mpas_xarray so it can remain an independent module, requiring
+minimal modification to accommodate MPAS-Analysis' needs.
+
+<h2> Requirement: MPAS-Analysis specific functionality should be supported in
+xarray preprossing <br>
+Date last modified: 2017/02/10 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+There should be a way to perform MPAS-Analysis specific functionality such as
+mapping variables during preprocessing.  This functionality should be
+relatively easy to add to as new preprocessing needs arise.
+
+
+<h1> Algorithmic Formulations (optional) </h1>
+
+<h2> Algorithm: mpas_xarray does not include MPAS-Analysis specific
+functionality <br>
+Date last modified: 2017/02/10 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+All functions and function arguments related to variable mapping will
+be removed from mpas_xarray and moved elsewhere.
+
+<h2> Algorithm: MPAS-Analysis specific functionality should be supported in
+xarray preprossing <br>
+Date last modified: 2017/02/15 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+A new utility function, `open_multifile_dataset` will added to `mpas_xarray` 
+that simplifies current calls to `xarray.open_mfdataset` to hide the 
+preprocessor and take care of removing redundant time indices once the dataset
+has been built. (This function doesn't directly address the requirement but
+is meant to make `mpas_xarray` easier to use and made sense because it
+has a one-to-one correspondence with other functionality, described below,
+that does address the requirement.)
+
+A new module, `generalized_reader` will also be added with its own
+`open_multifile_dataset` function.  This version takes additional arguments
+including a variable map and start and end dates for the dataset.
+`generalized_reader.open_multifile_dataset` will create a data set
+by calling `xarray.open_mfdataset` with its own preprocessing function,
+`generalized_reader._preprocess` that first maps variable names, then
+calls `mpas_xarray.preprocess` to finish the job.  Once the dataset has 
+been constructed, redundant time indices are removed and the 'Time'
+coordinate is sliced to be between the supplied start and end dates.
+
+This solution may add some confusion in terms of which reader should
+be used to open xarray datasets.  It is my sense that most developers
+adding new functionality will do so by modifying existing scripts, and
+these examples should make it clear which version of 
+`open_multifile_dataset` is most appropriate.  Nevertheless, clear
+documentation of `generalized_reader` and `mpas_xarray`, and their
+differences are needed.
```

Comment:
Can we please put some API code in the design document here to clarify usage, e.g., so a user doesn't have to dig through the existing code?